kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/core/v1/ServiceOperationsImpl.java

```diff
@@ -65,7 +61,10 @@ public Service waitUntilReady(long amount, TimeUnit timeUnit) {
     // if awaiting existence took very long, let's give at least 10 seconds to awaiting readiness
     long remaining = Math.max(10_000, timeUnit.toNanos(amount) - alreadySpent);
 
-    EndpointsOperationsImpl endpointsOperation = new EndpointsOperationsImpl(context);
+    HasMetadataOperation<Endpoints, EndpointsList, Resource<Endpoints>> endpointsOperation =
+        (HasMetadataOperation<Endpoints, EndpointsList, Resource<Endpoints>>) Handlers
+            .getOperation(Endpoints.class, EndpointsList.class, client, config)
+            .newInstance(context);
```

Comment:
Why the newInstance?
Doesn't `Handlers#getOperation` always return a new instance?

Reply:
It does return a new instances, but for the given http client and config.  Here you need an instance at a specific context - for which we already have a factory method for.

Reply:
Understood, thx!